addOns/pscanrulesBeta/src/test/java/org/zaproxy/zap/extension/pscanrulesBeta/InformationDisclosureSuspiciousCommentsUnitTest.java

```diff
@@ -0,0 +1,224 @@
+            assertFalse(msg.getResponseHeader().isJavaScript());
+            
+            // When
+            rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+            
+            // Then
+            assertEquals(1, alertsRaised.size());
+        }
+        
+        @Test
+        public void noSuspiciousCommentInElements() 
+        		throws HttpMalformedHeaderException, URIException {
+        	
+        	String body = "<h1>Some text <script>Some Html Element Fix: DO something </script></h1>\n"
+        			+ "<b>No script here</b>\n";
+            // Given
+            HttpMessage msg = createHttpMessageWithRespBody(body, "text/html;charset=ISO-8859-1");
+            
+            assertTrue(msg.getResponseHeader().isText());
+            assertFalse(msg.getResponseHeader().isJavaScript());
+            
+            // When
+            rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+            
+            // Then
+            assertEquals(0, alertsRaised.size());
+        }
+        
+        @Test
+        public void containsSuspiciousCommentInHTML() 
+        		throws HttpMalformedHeaderException, URIException {
+        	
+        	String body = "<h1>Some text <!--Some Html comment FixMe: DO something --></h1>\n"
+        			+ "<b>No script here</b>\n";
+            // Given
+            HttpMessage msg = createHttpMessageWithRespBody(body, "text/html;charset=ISO-8859-1");
+            
+            assertTrue(msg.getResponseHeader().isText());
+            assertFalse(msg.getResponseHeader().isJavaScript());
+            
+            // When
+            rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+            
+            // Then
+            assertEquals(1, alertsRaised.size());
+        }
+        
+        @Test
+        public void noSuspiciousCommentInHTML() throws HttpMalformedHeaderException, URIException {
+        	String body = "<h1>Some text <!--Some Html comment Fix: DO something --></h1>\n"
+        			+ "<b>No script here</b>\n";
+            // Given
+            HttpMessage msg = createHttpMessageWithRespBody(body, "text/html;charset=ISO-8859-1");
+            
+            assertTrue(msg.getResponseHeader().isText());
+            assertFalse(msg.getResponseHeader().isJavaScript());
+            
+            // When
+            rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+            
+            // Then
+            assertEquals(0, alertsRaised.size());
+        }
+
+        @Test
+        public void passesIfResponseIsEmpty() throws HttpMalformedHeaderException, URIException {
+            HttpMessage msg = createHttpMessageWithRespBody("", "text/html;charset=ISO-8859-1");
+
+            rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+            assertEquals(0, alertsRaised.size());
+        }
+
+        @Test
+        public void passesIfResponseIsNotText() throws HttpMalformedHeaderException, URIException {
```

Comment:
Could you make the test method names a bit more specific? (Refer to existing unit tests for inspiration.)

Obviously the unit test should "pass", this could be something more like `shouldNotAlertIfResponseIsNotText`